test/cpp/end2end/test_service_impl.cc

```diff
@@ -631,6 +636,8 @@ CallbackTestServiceImpl::BidiStream(CallbackServerContext* context) {
     bool setup_done_{false};
     std::thread finish_thread_;
     bool client_try_cancel_ = false;
+    std::promise<void> cancel_notifier_;
```

Comment:
prefer `absl::Notification` here, iirc `std::{promise,future}` are banned in various places

Reply:
We've been using `absl::Notification` in the server code for at about a year, so if that's the case there's more to clean up. I think that doc is being overly conservative here though.

Reply:
@veblush, can you confirm what the current state is of the absl synchronization library?  Is it actually okay to use gRPC at this point?